### DISCUSSION TITLE
GroupBy on relationship with same column name as parent

### DISCUSSION THREAD

**Discussion by igmltd** [laravel/framework#51764] [Q&A]
Hi, 

I have columns with the same name in both the parent table (Listings) and the related table (people), when i use groupBy it is using the 'area' column from the Parent rather than the relation, how can i get it to group by the relation 'area' column?

```
 $local_data = Listings::with('people')
                    ->where('email', $data['email'])
                    ->first();
                    
   $grouped_people = $local_data->people->groupBy('area');
```

Any help will be very much appreciated.

Thanks.

**Comment by chuckrincon** [comment]
Hey.

Try using the collection's `groupBy` function.

```php
$localData = Listings::with('people')
            ->where('email', $data['email'])
            ->get();

$groupedPeople = $localData->groupBy(fn (array $item) => $item['people']['area']);
```

**Reply by ghost** [reply]
@chuckrincon the get returns instances of Listings in this case, not arrays.
`$groupedPeople = $localData->groupBy(fn (Listings $item): int|float|string => $item['people']['area']);`

**Comment by wizzymore** [comment]
$grouped_people = $local_data->people->groupBy('area');

This is completely valid and works, no matter the columns of parent for the people relationship are.
